trieste/models/model_interfaces.py

```diff
@@ -311,6 +311,43 @@ def update(self, dataset: Dataset) -> None:
 
         self.model.data = dataset.query_points, dataset.observations
 
+    def covariance_between_points(
+        self, query_points_1: TensorType, query_points_2: TensorType
+    ) -> TensorType:
+        """
+        Compute the posterior covariance between sets of query points.
+
+        .. math:: \Sigma_{12} = K_{12} - K_{x1}(K_{xx} + \sigma^2 I)^{-1}K_{x2}
+
+        :param query_points_1: Set of query points with shape [N, D]
+        :param query_points_2: Sets of query points with shape [M, D]
+
```

Comment:
What happens if we have additional batch dimensions? i.e X_1 is [A, N, D] and X_2 is [B, M, D] ? I guess this is only really defined for A=B?

Reply:
The GPflow GPR model doesn't support leading batch dimensions, so these tensors need to have shapes `[N, D]` and `[M, D]` respectively.